python/cuml/decomposition/pca.pyx

```diff
@@ -514,20 +525,23 @@ class PCA:
         cdef uintptr_t singular_vals_ptr = self.singular_values_ptr
         cdef uintptr_t mean_ptr = self.mean_ptr
 
+        cdef cumlHandle* h_ = <cumlHandle*><size_t>self.handle.getHandle()
         if gdf_datatype.type == np.float32:
-            pcaInverseTransform(<float*> trans_input_ptr,
-                                      <float*> components_ptr,
-                                      <float*> singular_vals_ptr,
-                                      <float*> mean_ptr,
-                                      <float*> input_ptr,
-                                      params)
+            pcaInverseTransform(h_[0],
+                                <float*> trans_input_ptr,
+                                <float*> components_ptr,
+                                <float*> singular_vals_ptr,
+                                <float*> mean_ptr,
+                                <float*> input_ptr,
+                                params)
         else:
-            pcaInverseTransform(<double*> trans_input_ptr,
-                                      <double*> components_ptr,
-                                      <double*> singular_vals_ptr,
-                                      <double*> mean_ptr,
-                                      <double*> input_ptr,
-                                      params)
+            pcaInverseTransform(h_[0],
+                                <double*> trans_input_ptr,
+                                <double*> components_ptr,
+                                <double*> singular_vals_ptr,
+                                <double*> mean_ptr,
+                                <double*> input_ptr,
+                                params)
```

Comment:
Line 548 access `input_data` do we need to synchronize for this to work correctly?

Reply:
Same as above. Let me redirect the memcpy related question to @cjnolet

Reply:
with the sync call introduced above. This is not needed.

Reply:
This guarantee needs to come from the cudf side.